frontend/public/module/k8s/selector-requirement.js

```diff
@@ -8,42 +8,40 @@ export const createEquals = (key, value) => ({
   values: [value],
 });
 
+const validateKey = (key, operator, values) => {
+  // Need to validate key after parts split or else we don't take correct branch
+  if (/^(.*[0-9A-Za-z])$/.test(key)) {
```

Comment:
I don't think this regex is correct. `.*` matches any character, but not every character is allowed in a label key.

Reply:
https://kubernetes.io/docs/concepts/overview/working-with-objects/labels/#syntax-and-character-set

Reply:
Here is the error you get from the API server on an invalid label with the validation regex

> name part must consist of alphanumeric characters, '-', '_' or '.', and must start and end with an alphanumeric character (e.g. 'MyName',  or 'my.name',  or '123-abc', regex used for validation is '([A-Za-z0-9][-A-Za-z0-9_.]*)?[A-Za-z0-9]')" for field "metadata.labels".